src/main/java/org/springframework/cloud/deployer/spi/kubernetes/DefaultContainerFactory.java

```
@@ -137,7 +157,53 @@ protected Probe createProbe(Integer externalPort, String endpoint, int timeout,
+     *
+     * @param request AppDeploymentRequest - used to gather application overridden
+     * container command
+     * @return a list of strings that represents the command and any arguments for that command
+     */
+    private List<String> getContainerCommand(AppDeploymentRequest request) {
+        List<String> containerCommandList = new ArrayList<>();
+        String containerCommand = request.getDeploymentProperties()
+                .get("spring.cloud.deployer.kubernetes.containerCommand");
+        if (containerCommand != null) {
+			logger.info("Building container command from AppDeploymentRequest: "
+					+ containerCommand);
+            String[] containerCommandSplit = containerCommand.split(",");
+            for (String commandPart : containerCommandSplit) {
+                containerCommandList.add(commandPart.trim());
+            }
+        }
+        return containerCommandList;
+    }
+
+    /**
+     * @param request AppDeploymentRequest - used to gather additional container ports
+     * @return a list of Integers to add to the container
+     */
+    private List<Integer> getContainerPorts(AppDeploymentRequest request) {
+        List<Integer> containerPortList = new ArrayList<>();
+        String containerPorts = request.getDeploymentProperties()
+                .get("spring.cloud.deployer.kubernetes.containerPorts");
+        if (containerPorts != null) {
+            String[] containerPortSplit = containerPorts.split(",");
+            for (String containerPort : containerPortSplit) {
+                logger.info("Adding container ports from AppDeploymentRequest: "
+                        + containerPort);
+                try {
+                    Integer port = Integer.parseInt(containerPort.trim());
+                    containerPortList.add(port);
+                } catch (NumberFormatException e) {
+                    logger.info("Value is not a valid port number.");
```

Should not catch the error but let it bubble up